### ISSUE TITLE
Guest agent fails to restart when the VM restarts

### ISSUE THREAD

**Issue by IsaSih** [rancher-sandbox/rancher-desktop#6151]
### Actual Behavior

When creating a deployment and service via Kubernetes with containerd and switching to dockerd, the guest agent fails to start, causing an error in port forwarding

### Steps to Reproduce

1. Run Rancher Desktop with containerd
2. Create a deployment and a service: 
```
kubectl create deployment nginx-test --image=nginx:stable
kubectl create service clusterip nginx-test --tcp=5000:80
```
3. Switch to dockerd

### Result

Fatal error occurs:

```
'time="2023-12-07T13:44:01-08:00" level=info msg="[hostagent] Forwarding \\"/var/run/docker.sock\\" (guest) to \\"/Users/iguimaraes/.rd/docker.sock\\" (host)"\n' +
    'time="2023-12-07T13:45:01-08:00" level=info msg="[hostagent] Waiting for the final requirement 1 of 1: \\"boot scripts must have finished\\""\n' +
    'time="2023-12-07T13:45:01-08:00" level=info msg="[hostagent] The final requirement 1 of 1 is satisfied"\n' +
    'time="2023-12-07T13:45:01-08:00" level=error msg="[guest agent does not seem to be running; port forwards will not work]"\n' +
    'time="2023-12-07T13:45:01-08:00" level=warning msg="DEGRADED. The VM seems running, but file sharing and port forwarding may not work. (hint: see \\"/Users/iguimaraes/Library/Application Support/rancher-desktop/lima/0/ha.stderr.log\\")"\n' +
    'time="2023-12-07T13:45:01-08:00" level=fatal msg="degraded, status={Running:true Degraded:true Exiting:false Errors:[guest agent does not seem to be running; port forwards will not work] SSHLocalPort:60612}"\n',
  code: 1,
  [Symbol(child-process.command)]: '/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura start --tty=false 0'
}
```

[lima-logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13605837/lima-logs.zip)

[logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13605828/logs.zip)

### Expected Behavior

Application must restart with no errors

### Rancher Desktop Version

1.11.1-407-g4a132d0e

### Rancher Desktop K8s Version

1.27.7

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

macOS

### Operating System / Build Version

macOS Ventura 13.5.2

### What CPU architecture are you using?

arm64 (Apple Silicon)

### Linux only: what package format did you use to install Rancher Desktop?

None

**Comment by IsaSih** [comment]
Resetting Kubernetes fixes the issue

**Comment by IsaSih** [comment]
I came across the same error while testing #6159 
 When the VM restarts after snapshot cancellation, the fatal error occurs:

```
'time="2023-12-11T15:03:14-08:00" level=info msg="[hostagent] Forwarding \\"/var/run/docker.sock\\" (guest) to \\"/Users/iguimaraes/.rd/docker.sock\\" (host)"\n' +
    'time="2023-12-11T15:04:14-08:00" level=info msg="[hostagent] Waiting for the final requirement 1 of 1: \\"boot scripts must have finished\\""\n' +
    'time="2023-12-11T15:04:14-08:00" level=info msg="[hostagent] The final requirement 1 of 1 is satisfied"\n' +
    'time="2023-12-11T15:04:14-08:00" level=error msg="[guest agent does not seem to be running; port forwards will not work]"\n' +
    'time="2023-12-11T15:04:14-08:00" level=warning msg="DEGRADED. The VM seems running, but file sharing and port forwarding may not work. (hint: see \\"/Users/iguimaraes/Library/Application Support/rancher-desktop/lima/0/ha.stderr.log\\")"\n' +
    'time="2023-12-11T15:04:14-08:00" level=fatal msg="degraded, status={Running:true Degraded:true Exiting:false Errors:[guest agent does not seem to be running; port forwards will not work] SSHLocalPort:55433}"\n',
  code: 1,
  [Symbol(child-process.command)]: '/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura start --tty=false 0'
}
```

Logs: 

[logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13641896/logs.zip)
[lima-logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13641897/lima-logs.zip)

**Comment by IsaSih** [comment]
Error does not occur on 1.11

**Comment by ericpromislow** [comment]
I can't repro this on a non-ventura mac

**Comment by gaktive** [comment]
Jan not seeing on his various setups.

**Comment by jandubois** [comment]
I just reproduced this issue while testing on macOS 14.2 (Sonoma) on M1 (RD-173):

```
2023-12-13T18:03:50.988Z: Stopped Kubernetes backend cleanly.
2023-12-13T18:05:13.462Z: Kubernetes was unable to start: c [Error]: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima>
    at ChildProcess.<anonymous> (/Applications/Rancher Desktop.app/Contents/Resources/app.asar/dist/app/background.js:2:137484)
    at ChildProcess.emit (node:events:513:28)
    at Process.onexit (node:internal/child_process:291:12) {
  command: [
    '/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura',
    'start',
    '--tty=false',
    '0'
  ],
  stdout: '',
  stderr: 'time="2023-12-13T10:03:51-08:00" level=info msg="Using the existing instance \\"0\\""\n' +
    'time="2023-12-13T10:03:51-08:00" level=info msg="Starting the instance \\"0\\" with VM driver \\"qemu\\""\n' +
    'time="2023-12-13T10:03:51-08:00" level=info msg="QEMU binary \\"/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima>
    'time="2023-12-13T10:03:51-08:00" level=info msg="[hostagent] hostagent socket created at /Users/jan/Library/Application Support/rancher-de>
    'time="2023-12-13T10:03:51-08:00" level=info msg="[hostagent] Starting QEMU (hint: to watch the boot progress, see \\"/Users/jan/Library/Ap>
    'time="2023-12-13T10:03:51-08:00" level=info msg="SSH Local Port: 63025"\n' +
    'time="2023-12-13T10:03:51-08:00" level=info msg="[hostagent] Waiting for the essential requirement 1 of 2: \\"ssh\\""\n' +
    'time="2023-12-13T10:04:13-08:00" level=info msg="[hostagent] The essential requirement 1 of 2 is satisfied"\n' +
    'time="2023-12-13T10:04:13-08:00" level=info msg="[hostagent] Waiting for the essential requirement 2 of 2: \\"user session is ready for ss>
    'time="2023-12-13T10:04:13-08:00" level=info msg="[hostagent] The essential requirement 2 of 2 is satisfied"\n' +
    'time="2023-12-13T10:04:13-08:00" level=info msg="[hostagent] Waiting for the guest agent to be running"\n' +
    'time="2023-12-13T10:04:13-08:00" level=info msg="[hostagent] Forwarding \\"/var/run/docker.sock\\" (guest) to \\"/Users/jan/.rd/docker.soc>
    'time="2023-12-13T10:05:13-08:00" level=info msg="[hostagent] Waiting for the final requirement 1 of 1: \\"boot scripts must have finished\>
    'time="2023-12-13T10:05:13-08:00" level=info msg="[hostagent] The final requirement 1 of 1 is satisfied"\n' +
    'time="2023-12-13T10:05:13-08:00" level=error msg="[guest agent does not seem to be running; port forwards will not work]"\n' +
    'time="2023-12-13T10:05:13-08:00" level=warning msg="DEGRADED. The VM seems running, but file sharing and port forwarding may not work. (hi>
    'time="2023-12-13T10:05:13-08:00" level=fatal msg="degraded, status={Running:true Degraded:true Exiting:false Errors:[guest agent does not >
  code: 1,
  [Symbol(child-process.command)]: '/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl.ventura start --tty=>
}
```

**Comment by jandubois** [comment]
The VM seems to be running fine, but `containerd` has not been started:

```console
$ rdlimactl ls
NAME    STATUS     SSH                VMTYPE    ARCH       CPUS    MEMORY    DISK      DIR
0       Running    127.0.0.1:63025    qemu      aarch64    6       10GiB     100GiB    ~/Library/Application Support/rancher-desktop/lima/0
jan@studio:~/Library/Logs/rancher-desktop
$ rdlimactl shell 0 sudo rc-status
Runlevel: default
 lima-overlay                                                                      [  started  ]
 crond                                                                             [  started  ]
 lima-guestagent                                                       [  started 00:09:35 (0) ]
 lima-init-local                                                                   [  started  ]
 networking                                                                        [  started  ]
 acpid                                                                             [  started  ]
 procfs                                                                            [  started  ]
 qemu-binfmt                                                                       [  started  ]
 lima-init                                                                         [  started  ]
 sshd                                                                              [  started  ]
 udev-postmount                                                                    [  started  ]
Dynamic Runlevel: hotplugged
Dynamic Runlevel: needed/wanted
 sysfs                                                                             [  started  ]
 udev-settle                                                                       [  started  ]
 fsck                                                                              [  started  ]
 root                                                                              [  started  ]
 cgroups                                                                           [  started  ]
 localmount                                                                        [  started  ]
Dynamic Runlevel: manual
```

**Comment by jandubois** [comment]
The failure did not happen again when I repeated the exact same steps (RD-173):

* Factory Reset
* Accept default settings in First-Run dialog
* Change Settings: k8s 1.26.9, disable traffic, use `containerd`, Quit when closing main window
* Also set memory to 10GB, CPUs to 6, and mount type to 9p
* Press Accept

Here are the log files from the failure in the first attempt:

[degraded.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13664255/degraded.zip)

**Comment by jandubois** [comment]
And now I got the error on an Intel machine running 12.7.2 (Monterey) as well. When just starting up Rancher Desktop from unknown, but likely working previous state:

```console
$ cat background.log
2023-12-13T18:34:53.161Z: mainEvents settings-update: {"version":10,"application":{"adminAccess":false,"debug":false,"extensions":{"allowed":{"enabled":false,"list":[]},"installed":{}},"pathManagementStrategy":"rcfiles","telemetry":{"enabled":true},"updater":{"enabled":false},"autoStart":false,"startInBackground":false,"hideNotificationIcon":false,"window":{"quitOnClose":false}},"containerEngine":{"allowedImages":{"enabled":false,"patterns":[]},"name":"moby"},"virtualMachine":{"memoryInGB":6,"numberCPUs":2,"hostResolver":true},"WSL":{"integrations":{}},"kubernetes":{"version":"1.28.4","port":6443,"enabled":true,"options":{"traefik":true,"flannel":true},"ingress":{"localhostOnly":false}},"portForwarding":{"includeKubernetesServices":false},"images":{"showAll":true,"namespace":"k8s.io"},"containers":{"showAll":true,"namespace":"default"},"diagnostics":{"showMuted":false,"mutedChecks":{}},"experimental":{"virtualMachine":{"type":"qemu","useRosetta":false,"socketVMNet":false,"mount":{"type":"reverse-sshfs","9p":{"securityModel":"none","protocolVersion":"9p2000.L","msizeInKib":128,"cacheMode":"mmap"}},"networkingTunnel":false,"proxy":{"enabled":false,"address":"","password":"","port":3128,"username":"","noproxy":["0.0.0.0/8","10.0.0.0/8","127.0.0.0/8","169.254.0.0/16","172.16.0.0/12","192.168.0.0/16","224.0.0.0/4","240.0.0.0/4"]}}}}
2023-12-13T18:36:24.124Z: Kubernetes was unable to start: c [Error]: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
    at ChildProcess.<anonymous> (/Applications/Rancher Desktop.app/Contents/Resources/app.asar/dist/app/background.js:2:137484)
    at ChildProcess.emit (node:events:513:28)
    at Process.onexit (node:internal/child_process:291:12) {
  command: [
    '/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl',
    'start',
    '--tty=false',
    '0'
  ],
  stdout: '',
  stderr: 'time="2023-12-13T10:34:53-08:00" level=info msg="Using the existing instance \\"0\\""\n' +
    'time="2023-12-13T10:34:53-08:00" level=info msg="Starting the instance \\"0\\" with VM driver \\"qemu\\""\n' +
    'time="2023-12-13T10:34:53-08:00" level=info msg="QEMU binary \\"/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/qemu-system-x86_64\\" seems properly signed with the \\"com.apple.security.hypervisor\\" entitlement"\n' +
    'time="2023-12-13T10:34:54-08:00" level=info msg="[hostagent] hostagent socket created at /Users/jan/Library/Application Support/rancher-desktop/lima/0/ha.sock"\n' +
    'time="2023-12-13T10:34:54-08:00" level=info msg="[hostagent] Starting QEMU (hint: to watch the boot progress, see \\"/Users/jan/Library/Application Support/rancher-desktop/lima/0/serial*.log\\")"\n' +
    'time="2023-12-13T10:34:54-08:00" level=info msg="SSH Local Port: 62183"\n' +
    'time="2023-12-13T10:34:54-08:00" level=info msg="[hostagent] Waiting for the essential requirement 1 of 4: \\"ssh\\""\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] Waiting for the essential requirement 1 of 4: \\"ssh\\""\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] The essential requirement 1 of 4 is satisfied"\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] Waiting for the essential requirement 2 of 4: \\"user session is ready for ssh\\""\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] The essential requirement 2 of 4 is satisfied"\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] Waiting for the essential requirement 3 of 4: \\"sshfs binary to be installed\\""\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] The essential requirement 3 of 4 is satisfied"\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] Waiting for the essential requirement 4 of 4: \\"/etc/fuse.conf (/etc/fuse3.conf) to contain \\\\\\"user_allow_other\\\\\\"\\""\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] The essential requirement 4 of 4 is satisfied"\n' +
    'time="2023-12-13T10:35:22-08:00" level=info msg="[hostagent] Mounting \\"/Users/jan\\" on \\"/Users/jan\\""\n' +
    'time="2023-12-13T10:35:23-08:00" level=info msg="[hostagent] Mounting \\"/tmp/rancher-desktop\\" on \\"/tmp/rancher-desktop\\""\n' +
    'time="2023-12-13T10:35:23-08:00" level=info msg="[hostagent] Mounting \\"/Volumes\\" on \\"/Volumes\\""\n' +
    'time="2023-12-13T10:35:23-08:00" level=info msg="[hostagent] Mounting \\"/var/folders\\" on \\"/var/folders\\""\n' +
    'time="2023-12-13T10:35:23-08:00" level=info msg="[hostagent] Mounting \\"/private/tmp\\" on \\"/private/tmp\\""\n' +
    'time="2023-12-13T10:35:23-08:00" level=info msg="[hostagent] Mounting \\"/private/var/folders\\" on \\"/private/var/folders\\""\n' +
    'time="2023-12-13T10:35:23-08:00" level=info msg="[hostagent] Mounting \\"/Applications/Rancher Desktop.app/Contents/Resources/resources\\" on \\"/Applications/Rancher Desktop.app/Contents/Resources/resources\\""\n' +
    'time="2023-12-13T10:35:24-08:00" level=info msg="[hostagent] Waiting for the guest agent to be running"\n' +
    'time="2023-12-13T10:35:24-08:00" level=info msg="[hostagent] Forwarding \\"/var/run/docker.sock\\" (guest) to \\"/Users/jan/.rd/docker.sock\\" (host)"\n' +
    'time="2023-12-13T10:36:24-08:00" level=info msg="[hostagent] Waiting for the final requirement 1 of 1: \\"boot scripts must have finished\\""\n' +
    'time="2023-12-13T10:36:24-08:00" level=info msg="[hostagent] The final requirement 1 of 1 is satisfied"\n' +
    'time="2023-12-13T10:36:24-08:00" level=error msg="[guest agent does not seem to be running; port forwards will not work]"\n' +
    'time="2023-12-13T10:36:24-08:00" level=warning msg="DEGRADED. The VM seems running, but file sharing and port forwarding may not work. (hint: see \\"/Users/jan/Library/Application Support/rancher-desktop/lima/0/ha.stderr.log\\")"\n' +
    'time="2023-12-13T10:36:24-08:00" level=fatal msg="degraded, status={Running:true Degraded:true Exiting:false Errors:[guest agent does not seem to be running; port forwards will not work] SSHLocalPort:62183}"\n',
  code: 1,
  [Symbol(child-process.command)]: '/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl start --tty=false 0'
}
```

Full logs:

[logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13664448/logs.zip)

**Comment by jandubois** [comment]
Ran into the issue again (on the Intel machine). This time I have a `lima-guestagent.log`, but it doesn't show anything unexpected. Attaching log files here anyways, in case I need them again.

[logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13689433/logs.zip)

**Comment by IsaSih** [comment]
Found the same error while executing test RD-177. 
I created a snapshot via UI while trying to create a snapshot via CLI while the UI operation was still in progress. The CLI displayed an error message stating that "failed to create snapshot: backend lock file already exists; if there is no snapshot operation in progress, you can remove this error with `rdctl snapshot unlock`" But somehow, Rancher Desktop failed to restart with that guest agent errror.

Logs:
 
[logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13691134/logs.zip)
[lima-logs.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13691135/lima-logs.zip)

**Comment by jandubois** [comment]
@IsaSih The guest-agent log is stored inside the VM, so you need to copy it out to include it in the logs zipfile, e.g. with (quoting is important, `$HOME` must be substituted on the host and not in the VM):

```console
$ rdctl shell sudo sh -c "cp -v /var/log/lima-*.log $HOME/Library/Logs/rancher-desktop/"
'/var/log/lima-guestagent.log' -> '/Users/jan/Library/Logs/rancher-desktop/lima-guestagent.log'
'/var/log/lima-init.log' -> '/Users/jan/Library/Logs/rancher-desktop/lima-init.log'
```

**Comment by jandubois** [comment]
@IsaSih You should also permanently enable debug logging on your Mac, so the logs have additional information if the issue happens again. You can enable it (before you start Rancher Desktop) with

```console
launchctl setenv RD_DEBUG_ENABLED 1
```

This will then be in effect even after a factory reset (when you don't yet have a `settings.json`).

If you ever want to turn it off again:

```console
launchctl unsetenv RD_DEBUG_ENABLED
```

**Comment by jandubois** [comment]
Fixed by #6261
